src/psyclone/psyGen.py

```diff
@@ -562,9 +571,9 @@ def __init__(self, alg_invocation, idx, schedule_class, invokes,
         self._schedule = schedule_class(self._name, alg_invocation.kcalls,
                                         reserved_names)
 
-        # TODO #1170: Must be fixed before uncommenting the code below
-        # if self.invokes:
-        #     self.invokes.psy.container.addchild(self._schedule)
+        # Add the new Schedule in a top-level PSy Container
+        if self.invokes:
+            self.invokes.psy.container.addchild(self._schedule)
```

Comment:
So we already had the Container node created?

Reply:
Yes, this was done for generating the PSy-layer using the PSyIR backend on the gocean/eg6 but left commented because of the issues that this PR solves.